webapp/channels/src/components/admin_console/data_retention_settings/global_policy_form/global_policy_form.tsx

```diff
@@ -90,16 +108,16 @@ export default class GlobalPolicyForm extends React.PureComponent<Props, State>
 
         newConfig.DataRetentionSettings.EnableMessageDeletion = this.setDeletionEnabled(messageRetentionDropdownValue.value);
 
-        const messageDays = this.setRetentionDays(messageRetentionDropdownValue.value, messageRetentionInputValue);
-        if (messageDays >= 1) {
-            newConfig.DataRetentionSettings.MessageRetentionDays = messageDays;
+        if (!this.isMessageRetentionSetByEnv()) {
```

Comment:
Why is there a special case for environment variables? Isn't the setting disabled anyway if an env is used?

Reply:
And if need this special case: Should days be set to 0 here and the config file only uses hours moving forward?

Reply:
> Why is there a special case for environment variables? Isn't the setting disabled anyway if an env is used?

There can be cases where maybe `messageRetention` is set by an env variable but `fileRetention` isn't, I didn't want to silently overwrite whatever is in their config.json without them knowing when they click save on this page. Very niche edge case but I think it's necessary.

> Should days be set to 0 here and the config file only uses hours moving forward?

Yeah that makes sense, I'll change this to only update hours

Reply:
Done